specs/network_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,131 @@
+= Specification for adding network details in node hardware inventory
+
+Add network details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently network details are missing in node hardware inventory.
+we have to add network details like IPv4, IPv6, netmask, status as a part of
+node hardware inventory.
+
+== Use Cases
+
+Tendrl should show the current status of active network interface.
+
+== Proposed change
+
+* From node_agent add a new function in pull_hardware_inventory to find active
+  network interfaces details except lo, virbr0, virbr0-nic and update those
+  details into node_inventory list.
+* Create a new class in persistence module to keep network details.
+* Add a new function in persister to persist network details.
+* Modify the manager to populate the instance of newly created class using
+  node_inventory list and persist the instance in etcd.
+
+== Alternatives
+
+* Netifaces package is used to find active network interface in particular
+  node.
+* Add netifaces package as dependency.
+
+== Data model impact
+
+Add network details in etcd with key nodes/node_id/Network/interface_name
+
+== Impacted modules
+
+=== Tendrl/common impact
+
+None
+
+=== Tendrl/node_agent impact
+
+* Add a new function in pull_hardware_inventory to collect active network
+  interface details.
+* Create a new class in persistence module to keep the network details.
+* Add a new function in persister to persist network details.
+* Modify the manager modules to read and store active network interface details.
+* Commands used to retrieve network details are:
+```
+    interfaces = netifaces.interfaces()
+    ipv4_addr_list = netifaces.ifaddresses(interfaces[integer_value])[AF_INET]
+    ipv6_addr_list = netifaces.ifaddresses(interfaces[integer_value])[AF_INET6]
+    netmask = ipv4_addr_detail[integer_value]['netmask']
+```
+* Excpected output format for the active network details is:
+```
+{"interface_name": {"ipv4": ["ipv4address", ...],
+                    "ipv6": ["ipv6address", ...],
+                    "netmask": ["subnet", ...]
+                    "status":"up/down"
+                   }, ...
+}
+```
+
+
+=== Tendrl/gluster_integration impact
+
+None
+
+=== Tendrl/ceph_integration impact
+
+None
+
+== Security impact
+
+None
+
+== Notifications/Monitoring impact
+
+We might have to monitor the status of the network interface(up/down).
```

Comment:
I would certainly prefer the option-2 of having a collectd plugin to push data to graphite (if we need, for sure I prefer this to be there in graphite. The only thing we need to see if real estate available for the same in dashboard)